client/src/actions/bulk.js

```diff
@@ -73,5 +80,58 @@ export function bulkValidateSuggestions(form) {
 }
 
 export function bulkCreateSuggestions(form) {
-    return doAction(actions.create, form);
+    return doAction(actions.createSuggestion, form, true);
+}
+
+export function bulkCreateMonuments(form) {
+    return async dispatch => {
+        dispatch(pending(actions.create));
+
+        try {
+            // Create the BulkCreateMonumentSuggestion
+            let suggestResult = await (await fetch(actions.create.suggestUri, {
+                method: 'post',
+                body: buildFormData(form)
+            })).json();
+
+            const suggestJobId = suggestResult.id;
+
+            let interval;
+            await new Promise(resolve => {
+                interval = window.setInterval(async () => {
+                    suggestResult = await (await fetch(`${actions.create.suggestUri}/progress/${suggestJobId}`)).json();
+                    // Divide the suggestion progress by 2 so it ranges from 0.0 to 0.50
+                    dispatch(pending(actions.create, (suggestResult.progress / 2)));
+
+                    if (suggestResult.future && suggestResult.future.done) resolve();
+                }, 200);
+            });
+            window.clearInterval(interval);
+
+            suggestResult = await (await fetch(`${actions.create.suggestUri}/result/${suggestJobId}`)).json();
+
+            // Approve the BulkCreateMonumentSuggestion
+            let approveResult = await (await fetch(`${actions.create.approveUri}/${suggestResult.id}/approve`, {
+                method: 'put'
+            })).json();
+
+            const approveJobId = approveResult.id;
+
+            await new Promise(resolve => {
+                interval = window.setInterval(async () => {
+                    approveResult = await (await fetch(`${actions.create.approveUri}/approve/progress/${approveJobId}`)).json();
+                    // Divide the approve progress by 2 then add 0.50 so it ranges from 0.50 to 1.00
+                    dispatch(pending(actions.create, ((approveResult.progress / 2) + 0.50)));
+
+                    if (approveResult.future && approveResult.future.done) resolve();
+                }, 200);
+            });
+            window.clearInterval(interval);
+
+            approveResult = await (await fetch(`${actions.create.approveUri}/approve/result/${approveJobId}`)).json();
+            dispatch(success(actions.create, approveResult));
+        } catch (err) {
+            dispatch(error(actions.create, err));
+        }
+    }
```

Comment:
I tried for awhile to find a way to make the logic for consuming an `AsyncJob` API generic but I couldn't find a clean way to do it.

Reply:
`doAction` returns its own `async dispatch`. If I wanted to do what we were doing in `bulkCreateMonuments` wouldn't I need to pass `dispatch` down as a parameter? Can I do that?